### PR TITLE
Add dependency information and makefile to build/update virtualenv.

### PR DESCRIPTION
**Pull request by anarazel**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/venv/
+/.ipynb_checkpoints/
+/prefetch.log
```
Comment (melanieplageman) [inline .gitignore:3]: I took out prefetch log in the last refactor but I will probably add it back in so leaving this makes sense for now to me :)

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# default to sdist target
+all: sdist
```
Comment (melanieplageman) [inline Makefile:2]: is it okay to assume people have virtualenv installed?

File: README.md [hunk 3]
```diff
@@ -1 +1,10 @@
 # prefetch_modeler
+
+To create a python venv with the required dependencies run
+```shell
+make sdist
+```
+To start the jupyter notebook server run
+```shell
+make run-notebook
```
Comment (melanieplageman) [inline README.md]: should be `start-notebook`, no?

File: .gitignore (hunk 1 above)
Comment (anarazel) [reply .gitignore:3]: Heh, happy to take it out as well...

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,254 @@
+#
+# This file is autogenerated by pip-compile with python 3.9
+# To update, run:
+#
+#    pip-compile
```
Comment (melanieplageman) [inline requirements.txt:5]: isn't this a superset of what is required for this project?

File: Makefile (hunk 2 above)
Comment (anarazel) [reply Makefile:2]: I think so. It's definitely less of an assumption than having all of jupyter pandas etc installed? If this gets more widely used we can add a setup.cfg or such

File: requirements.txt (hunk 4 above)
Comment (anarazel) [reply requirements.txt:5]: What is superfluous? The direct requirements, in requirements.in, are I think the minimal ones? I think without each of the four listed there I saw failures?

**Review by melanieplageman — CHANGES_REQUESTED** [review]
thanks for doing this! we can leave all of my "questions/suggestions" for now except I think the README has the wrong make target listed for the jupyter notebook. I'll add you to collaborators and then you can push this.

File: README.md (hunk 3 above)
Comment (anarazel) [reply README.md]: Fixed. Gah... :)

**Comment by anarazel** [comment]
> thanks for doing this! we can leave all of my "questions/suggestions" for now except I think the README has the wrong make target listed for the jupyter notebook. I'll add you to collaborators and then you can push this.

Cool! I've fixed the README thing now. I don't know enough github to know if I can merge after the collaborator changes after you requested changes...

**Comment by anarazel** [comment]
Merged, thanks!

**Pull request merged**